library_generation/model/generation_config.py

```diff
@@ -61,6 +63,15 @@ def get_proto_path_to_library_name(self) -> dict[str, str]:
     def is_monorepo(self) -> bool:
         return len(self.libraries) > 1
 
+    def contains_common_protos(self) -> bool:
```

Comment:
I added this method to determine whether there are common protos in the configuration.

We can't just have one `is_gapic_monorepo` method since we have three cases:
- sdk-platform-java, `contains_common_protos` returns `True`
- google-cloud-java, `contains_common_protos` returns `False` and `is_monorepo` returns `True`
- handwritten libraries, `contains_common_protos` returns `False` and `is_monorepo` returns `False`

Reply:
I removed the setter method.